src/js/components/Drop/stories/Simple.js

```diff
@@ -1,36 +1,118 @@
+  const [showDropInModal, setShowDropInModal] = useState(false);
+  const [showRestrictFocusDrop, setShowRestrictFocusDrop] = useState(false);
+  const layerRef = useRef();
+  const buttonRef = useRef();
+  const layerButtonRef = useRef();
+  const restrictFocusButtonRef = useRef();
 
-  const [, setShowDrop] = useState(false);
-  useEffect(() => {
-    setShowDrop(true);
-  }, []);
+  const onClose = () => setShowLayer(false);
 
   return (
     // Uncomment <Grommet> lines when using outside of storybook
     // <Grommet theme={...}>
-    <Box fill align="center" justify="center">
-      <Box
-        background="dark-2"
-        pad="medium"
-        align="center"
-        justify="start"
-        ref={targetRef}
-      >
-        Target
+    <>
+      <Box direction="row" pad="large" gap="small">
+        <DropButton
+          label="Drop button"
+          dropContent={<InteractiveContent />}
+          dropAlign={align}
+        />
+        <Button
+          ref={buttonRef}
+          label="Regular button"
+          onClick={() => setShowDrop(true)}
+        />
+        <Button
+          ref={layerRef}
+          label="Open layer"
+          onClick={() => setShowLayer(true)}
+        />
+        {showLayer && (
+          <Layer
+            position="right"
+            full="vertical"
+            modal
+            onClickOutside={onClose}
+            onEsc={onClose}
+          >
+            <Box
+              fill="vertical"
+              overflow="auto"
+              width="medium"
+              pad="medium"
+              gap="small"
+            >
+              {/* <Anchor href="#" label="testing anchor" /> */}
+              <DropButton
+                label="Another drop button"
+                dropContent={<InteractiveContent />}
+                dropAlign={align}
+              />
+              <Button
+                ref={layerButtonRef}
+                label="Another regular button"
+                onClick={() => setShowDropInModal(true)}
+              />
+              {showDropInModal && (
+                <Drop
+                  target={layerButtonRef.current}
+                  align={align}
+                  onClickOutside={() => setShowDropInModal(false)}
+                  onEsc={() => setShowDropInModal(false)}
+                >
+                  <InteractiveContent />
+                </Drop>
+              )}
+              <Button
+                ref={restrictFocusButtonRef}
+                label="A regular button with restrict focus"
+                onClick={() => setShowDropInModal(true)}
+              />
+              {showRestrictFocusDrop && (
+                <Drop
+                  target={restrictFocusButtonRef.current}
+                  align={align}
+                  onClickOutside={() => setShowRestrictFocusDrop(false)}
+                  onEsc={() => setShowRestrictFocusDrop(false)}
+                  // TO DO onEsc isn't restoring
```

Comment:
just saw this locally adding comment so I remember to look into this more as well!

Reply:
The 'original element' is coming back as the whole `body` in the dom so the focus is not going back on the `button` thats not the correct 'original element'

Reply:
This is resolved now, realized it was actually an error in how I'd implemented the story.